core/src/main/java/org/ehcache/events/CacheEventNotificationServiceImpl.java

```diff
@@ -252,4 +282,28 @@ public void setSource(Cache<K, V> source) {
     }
   }
 
+  private static class EventThreadLocal {
+    private final ThreadLocal<List<CacheEventWrapper>> threadLocal = new ThreadLocal<List<CacheEventWrapper>>() {
+      @Override
+      protected List<CacheEventWrapper> initialValue() {
+        return new ArrayList<CacheEventWrapper>();
+      }
+    };
+
+    public List<CacheEventWrapper> get() {
+      HashMap<Object, CacheEventWrapper> cacheEventMap = new HashMap<Object, CacheEventWrapper>();
+      for (CacheEventWrapper cacheEventWrapper : threadLocal.get()) {
+        if((cacheEventWrapper.cacheEvent.getType() == EventType.CREATED)
```

Comment:
What are you trying to achieve here?
Because you are creating a map, iterating over elements, all for removing duplicates ... but at each call to `get()`!!